src/App/LoadFigure.purs

```diff
@@ -46,22 +50,28 @@ figSpecFromJson spec@{ datasets, file, imports, inputs, query, linking } =
    , linking
    }
 
-loadFigure :: String -> Effect Unit
-loadFigure fileName = runAffs_ (uncurry drawFig)
-   [ do
-        -- TODO: simplify
-        result <- get json fileName
-        case result of
-           Left err -> error ("Json fetching failed with " <> printError err)
-           Right response ->
-              case decodeJson response.body of
-                 Left err -> error ("JSON decoding failed with " <> show err)
-                 Right spec -> do
-                    let spec'@{ fluidSrcPaths } = figSpecFromJson spec
-                    ("fig" × _) <$> runWebT (FileCxt { fluidSrcPaths }) (loadFig spec')
+loadSpec :: String -> Aff Json
+loadSpec filename = do
+   result <- get json filename
+   case result of
+      Left err -> throwError (error ("Json fetching failed with " <> printError err))
+      Right response -> pure $ response.body
+
+loadFigureFromJson :: Json -> Effect Unit
+loadFigureFromJson json = runAffs_ (uncurry drawFig)
+   [ case decodeJson json :: Either JsonDecodeError JsonSpec of
+        Left err -> throwError (error ("JSON decoding failed with " <> show err))
+        Right spec -> ("fig" × _) <$> runWebT (FileCxt { fluidSrcPaths }) (loadFig figSpec)
+           where
+           figSpec@{ fluidSrcPaths } = figSpecFromJson spec
    ]
 
+loadFigure :: String -> Effect Unit
+loadFigure filename = launchAff_ do
+   jsonSpec <- loadSpec filename
+   liftEffect $ loadFigureFromJson jsonSpec
+
 drawCode :: String -> String -> Effect Unit
 drawCode folder file = runAffs_ drawFile
    [ runWebT (FileCxt { fluidSrcPaths: [ Folder folder ] }) $ loadFile' [ Folder folder ] (File file)
-   ]
+   ]
```

Comment:
Every source file needs to end in a closing line terminator -- set up VSCode to do this automatically: https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files